### ISSUE TITLE
blockmedian and blockmode does not complain if -S is used

### ISSUE THREAD

**Issue by anbj** [GenericMappingTools/gmt#3455]
I wanted to use `-Sn -G<grd>` with `blockmedian`, but realised after running it, it does not offer that option. It made a grid and apparently all was good. Maybe throw a warning or error if the users tries to do `-S[..]` on `blockmedian` and `blockmean`? Why is `-S` only available for `blockmean`?

**Comment by PaulWessel** [comment]
You are right, no error if -S is used.  how can that be? Will check. Same with -Z -m or other random letters...

**Comment by anbj** [comment]
Did not like this though

```
$ gmt blockmedian xyz_dump_l44-99.gmt $R -V -I20e -Sn -Gcov.grd -Dingleberry
blockmedian [ERROR]: Unrecognized option -D
```

**Comment by PaulWessel** [comment]
Tied into old GMT 4 compatibility settings.  Opening a separate issue on this.

**Comment by PaulWessel** [comment]
I am closing this issues since it will be addressed in a manner determined by the issue in #3456.
